src/ImageRunModal.jsx

```diff
@@ -28,88 +28,101 @@ const units = {
     },
 };
 
-class PublishPortsForm extends React.Component {
+const PublishPort = (props) => {
+    return (
+        <React.Fragment>
+            <div role='group' className='ct-form-layout-split'>
+                <input className='form-control'
+                       id={props.id}
+                       type='number'
+                       step={1}
+                       min={1}
+                       max={65535}
+                       placeholder={_("Container port")}
+                       value={props.item.containerPort || ''}
+                       onChange={e => props.onChange(props.idx, 'containerPort', e.target.value)} />
+                <Select.Select className='form-control'
+                               initial={props.item.protocol}
+                               onChange={value => props.onChange(props.idx, 'protocol', value)} >
+                    <Select.SelectEntry data='TCP' key='TCP'>
+                        {_("TCP")}
+                    </Select.SelectEntry>
+                    <Select.SelectEntry data='UDP' key='UDP'>
+                        {_("UDP")}
+                    </Select.SelectEntry>
+                </Select.Select>
+                <input className='form-control'
+                       type='number'
+                       step={1}
+                       min={1}
+                       max={65535}
+                       placeholder={_("Host port")}
+                       value={props.item.hostPort || ''}
+                       onChange={e => props.onChange(props.idx, 'hostPort', e.target.value)} />
+            </div>
+            <div role='group' className='ct-form-layout-split'>
+                <Button bsStyle='default' className='pficon-close'
+                        disabled={ props.idx === 0 && !props.item.hostPort && !props.item.containerPort }
+                        onClick={() => props.removeitem(props.idx)} />
+                <Button bsStyle='default' className='fa fa-plus' onClick={props.additem} />
+            </div>
+        </React.Fragment>
+    );
+};
+
+class DynamicListForm extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            publish: [{ key: 0, containerPort: null, hostPort: null, protocol: 'TCP' }],
+            list: [Object.assign({ key: 0 }, props.default, {})],
```

Comment:
All your `Object.assign()` calls now have an extra `{}` at the end. This seems to be a no-op, or am I missing anything?

Reply:
yup you're right, this doesn't do anything. I'll remove it

Reply:
Please actually remove them then :-)